### PR TITLE
Enable users to choose preferred language

### PR DESCRIPTION
**Pull request by simoneduca**
Describe your changes.
Add a dropdown for users to select the languages available
Stores users preference for next visits
# Review Checklist

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [x] Did you deploy a staging branch? https://add-locales-menu.pfe-preview.zooniverse.org/
- [ ] Did you get any type checking errors from [Babel Typecheck](https://github.com/codemix/babel-plugin-typecheck)

## Optional

- [ ] If it's a new component, is it in ES6? Is it clear of warnings from ESLint?
- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?

### CODE REVIEW

File: app/lib/language-engine.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import counterpart from 'counterpart';
+
+const load = (language) => {
+  const defaultLanguage = { label: 'English', value: 'en' };
```
Comment (eatyourgreens) [inline app/lib/language-engine.js]: Should this default to the project primary language if the project isn't in English?

File: app/lib/language-selector.jsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import React, { Component, PropTypes } from 'react';
+import Select from 'react-select';
+import languages from './languages';
+import { select } from './language-engine';
+
+const propTypes = {
+  props: PropTypes.object.isRequired
+};
+const storedLanguage = JSON.parse(localStorage.getItem('preferred-language'));
+const defaultLanguage = { label: 'English', value: 'en' };
```
Comment (eatyourgreens) [inline app/lib/language-selector.jsx]: default language is defined in more than one place, which could make it harder to change.

File: app/lib/language-selector.jsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
+import React, { Component, PropTypes } from 'react';
+import Select from 'react-select';
```
Comment (eatyourgreens) [inline app/lib/language-selector.jsx]: Why react-select instead of a dropdown menu?

**Comment by simoneduca** [comment]
@eatyourgreens given that this a fairly self-contained feature, I was thinking of using Redux, like Feedback does.

File: app/lib/language-engine.js (hunk 1 above)
Comment (simoneduca) [inline app/lib/language-engine.js]: Good point, yes. As you said, that would accommodate people creating projects with a different default language than English.

**Comment by eatyourgreens** [comment]
When the language changes, the [document language](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/lang) should be changed in the HTML too.

Then we can do nifty things like [styling Hebrew as RTL](/zooniverse/Galaxy-Zoo/blob/master/css/locales.styl#L26-L27).

**Comment by simoneduca** [comment]
Neat

**Comment by eatyourgreens** [comment]
There's [a request on Talk for RTL language support](https://www.zooniverse.org/talk/18/392214?comment=655055).

**Comment by simoneduca** [comment]
@eatyourgreens This is so old it's probably safe to close it and start again if this feature is required in the future. Is that ok with you?

**Comment by simoneduca** [comment]
I take that as a 'yes'. Closing.

**Pull request closed without merging**